snap/naming/componentref.go

```diff
@@ -35,6 +35,16 @@ func NewComponentRef(snapName, componentName string) ComponentRef {
 	return ComponentRef{SnapName: snapName, ComponentName: componentName}
 }
 
+// SplitFullComponentName splits <snap>+<comp> in <snap> and <comp> strings.
+func SplitFullComponentName(fullComp string) (string, string, error) {
+	names := strings.Split(fullComp, "+")
```

Comment:
```suggestion
	names := strings.SplitN(fullComp, "+", 2)
```
or even better strings.Cut()

Reply:
I left this with Split, as SplitN and Cut would not detect if you have more than one `+`, thing that we do not allow.